pymilvus/milvus_client/_utils.py

```diff
@@ -1,11 +1,49 @@
+import asyncio
 import hashlib
 import logging
 
+from pymilvus.client.async_grpc_handler import AsyncGrpcHandler
+from pymilvus.client.grpc_handler import GrpcHandler
 from pymilvus.orm.connections import connections
 
 logger = logging.getLogger(__name__)
 
 
+def _check_async_connection_event_loop(using: str) -> bool:
+    """Check if the async connection's event loop matches the current running loop.
+
+    Returns:
+        bool: True if event loop matches, False otherwise
+    """
+    try:
+        handler = connections._fetch_handler(using)
+        if handler is None or not isinstance(handler, AsyncGrpcHandler):
+            return False
+
+        if handler._async_channel is None:
+            return False
+
+        try:
+            current_loop = asyncio.get_running_loop()
+        except RuntimeError:
+            return False
+
+        # Check if channel's event loop matches current loop
+        channel = handler._async_channel
+        cython_ch = channel._cython_channel
+        if cython_ch is None:
+            return False
+
+        channel_loop = cython_ch._loop
```

Comment:
private vars may change  by grpc in anytime, it's not always safe. And why are we checking them?